x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/FormatterState.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.sql.plugin;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xpack.sql.action.BasicFormatter;
+
+import java.io.IOException;
+
+public record FormatterState(BasicFormatter formatter) implements Writeable {
```

Comment:
Not sure wether this class is a bit of an overkill right now but I added it for additional clarity to avoid passing `BasicFormatter` everywhere. Also, it could act as an extension point in the future in case we have to add additional state for other formatters.